app/src/main/java/fr/neamar/kiss/SettingsActivity.java

```diff
@@ -50,6 +53,34 @@ protected void onCreate(Bundle savedInstanceState) {
         addExcludedAppSettings(prefs);
 
         addSearchProvidersSelector(prefs);
+
+        addSmsApplicationsSelector(prefs);
+    }
+
+    private void addSmsApplicationsSelector(SharedPreferences prefs) {
+        ListPreference smsAppList = (ListPreference) findPreference("sms-apps");
+        Intent intent = new Intent(Intent.ACTION_SENDTO);
+        intent.setType("text/plain");
+        intent.setData(Uri.parse("sms:"));
+        final PackageManager pm = getPackageManager();
+        List<ResolveInfo> availableSmsApps = pm.queryIntentActivities(intent, PackageManager.MATCH_DEFAULT_ONLY);
+
+        CharSequence[] entries = new CharSequence[availableSmsApps.size() + 1];
+        CharSequence[] entryValues = new CharSequence[availableSmsApps.size() + 1];
+
+        entries[0] = this.getString(R.string.sms_apps_default);
+        entryValues[0] = "defaultSmsApp";
+        int i = 1;
+        for (ResolveInfo info : availableSmsApps) {
+            String appName = (String) (info.activityInfo.applicationInfo != null ? pm.getApplicationLabel(info.activityInfo.applicationInfo) : "(unknown)");
```

Comment:
Please make the string "(unknown)" translatable (it might be shown to the user after all)